src/vs/workbench/contrib/terminal/common/scripts/shellIntegration-bash.sh

```diff
@@ -214,16 +217,77 @@ __vsc_update_cwd() {
 	builtin printf '\e]633;P;Cwd=%s\a' "$(__vsc_escape_value "$__vsc_cwd")"
 }
 
-# __vsc_update_env() {
-# 	builtin printf '\e]633;EnvSingleStart;%s;\a' $__vsc_nonce
-# 	for var in $(compgen -v); do
-# 		if printenv "$var" >/dev/null 2>&1; then
-# 			value=$(builtin printf '%s' "${!var}")
-# 			builtin printf '\e]633;EnvSingleEntry;%s;%s;%s\a' "$var" "$(__vsc_escape_value "$value")" $__vsc_nonce
-# 		fi
-# 	done
-# 	builtin printf '\e]633;EnvSingleEnd;%s;\a' $__vsc_nonce
-# }
+
+updateEnvCache() {
+	local key="$1"
+	local value="$2"
+
+	for i in "${!vsc_env_keys[@]}"; do
+		if [[ "${vsc_env_keys[$i]}" == "$key" ]]; then
+			if [[ "${vsc_env_values[$i]}" != "$value" ]]; then
```

Comment:
this allow us to only printf and send to VS Code if there is a difference in terms of value.

`builtin printf '\e]633;EnvSingleEntry;%s;%s;%s\a' "$key" "$(__vsc_escape_value "$value")" "$__vsc_nonce"`
below allow us to send to VS Code if it is a brand new variable.